File: vehicle-rentals-microservices/bookings-service/src/main/java/com/project/bookings_service/entity/Booking.java
package com.project.bookings_service.entity;

import java.time.LocalDateTime;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
public class Booking {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long bookingId;
	private LocalDateTime bookingDate;
	private LocalDateTime fromDate;
	private LocalDateTime toDate;
	private double duartionHours;
	private double price;
	private long userId;
	private long vehicleId;
	private String status;
	@PrePersist
	public void UpdateDate() {
		this.bookingDate=LocalDateTime.now();
	}
	
}

File: onthego/src/app/models/booking.ts
export class Booking {
    bookingId: number;
    bookingDate: Date;
    fromDate: Date;
    toDate: Date;
    durationHours: number;
    price: number;
    userId: number;
    vehicleId: number;
    status: string;

    constructor(
        bookingId: number = 0,
        bookingDate: Date = new Date(),
        fromDate: Date,
        toDate: Date,
        durationHours: number,
        price: number,
        userId: number,
        vehicleId: number,
        status: string
    ) {
        this.bookingId = bookingId;
        this.bookingDate = bookingDate;
        this.fromDate = fromDate;
        this.toDate = toDate;
        this.durationHours = durationHours;
        this.price = price;
        this.userId = userId;
        this.vehicleId = vehicleId;
        this.status = status;
    }

    updateBookingDate(): void {
        this.bookingDate = new Date();
    }
}
